`azurerm_data_factory_linked_service_key_vault`- add `key_vault_base_url_dynamic_expression` attribute to support dynamic expressions for Key Vault base URL (#30026)
hashicorp/terraform-provider-azurerm: website/docs/r/data_factory_linked_service_key_vault.html.markdown

* `data_factory_id` - (Required) The Data Factory ID in which to associate the Linked Service with. Changing this forces a new resource.

* `key_vault_id` - (Optional) The ID the Azure Key Vault resource. Mutually exclusive with `key_vault_base_url_dynamic_expression`. Use `key_vault_base_url_dynamic_expression` instead if the URL is a dynamic expression.

> * `key_vault_base_url_dynamic_expression` - (Optional) The base url of the Azure Key Vault. Use only if the base url is a dynamic expression (for example `@{concat('https://my-kv-',linkedService().environmentName,'.vault.azure.net')}`). Use `key_vault_id` otherwise. This attribute is mutually exclusive with `key_vault_id`.

Reviewer: ```suggestion
* `key_vault_base_url_dynamic_expression` - (Optional) The base url of the Azure Key Vault. Use only if the base url contains dynamic expressions (for example `@{concat('https://my-kv-',linkedService().environmentName,'.vault.azure.net')}`). Use `key_vault_id` otherwise. This attribute is mutually exclusive with `key_vault_id`.
```

Minor grammatical improvement

Author: Made some additional changes to `key_vault_id`'s documentation for consistency
